### PR TITLE
create-dmg/x.x.x: change support location

### PR DESCRIPTION
**Pull request by MartinDelille**
Specify library name and version:  **create-dmg**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/create-dmg/all/conanfile.py [hunk 1]
```diff
@@ -13,24 +13,32 @@ class CreateDmgConan(ConanFile):
     homepage = "https://github.com/create-dmg/create-dmg"
     url = "https://github.com/conan-io/conan-center-index"
     settings = "os", "arch", "compiler", "build_type"
-    no_copy_source = True
 
     @property
     def _source_subfolder(self):
         return "source_subfolder"
 
+    def export_sources(self):
```
Comment (uilianries) [inline recipes/create-dmg/all/conanfile.py:21]: revert. It's a tool package. https://docs.conan.io/en/latest/creating_packages/existing_binaries.html#downloading-and-packaging-pre-built-binaries

Comment (MartinDelille) [reply recipes/create-dmg/all/conanfile.py:21]: I did this in order to be able to patch `create-dmg` to properly access to the `support` directory: https://github.com/conan-io/conan-center-index/pull/11569/files#diff-70f9e2e6516706b1b866dc40b6094b688fa797d9d698871b83399d82e54e371e

Comment (MartinDelille) [reply recipes/create-dmg/all/conanfile.py:21]: @uilianries How can I properly patch a tool package ?

Comment (uilianries) [reply recipes/create-dmg/all/conanfile.py:21]: Follow the pattern that is showed in the link. Any patch should be applied on build step as always.

**Comment by MartinDelille** [comment]
@uilianries Unfortunately, when removing the export_sources and source method, the patch file is innaccessible:

```
conan export create-dmg/all create-dmg/1.1.0@
Exporting package recipe
create-dmg/1.1.0 exports: File 'conandata.yml' found. Exporting it...
create-dmg/1.1.0 exports: Copied 1 '.yml' file: conandata.yml
create-dmg/1.1.0: A new conanfile.py version was exported
create-dmg/1.1.0: Folder: /Users/martin/.conan/data/create-dmg/1.1.0/_/_/export
create-dmg/1.1.0: Package recipe modified in export, forcing source folder removal
create-dmg/1.1.0: Use the --keep-source, -k option to skip it
create-dmg/1.1.0: Exported revision: e256052a9a1cbf1c8ceac235ac126877
conan test create-dmg/all/test_package create-dmg/1.1.0 -b create-dmg --profile Darwin.txt
Configuration:
[settings]
arch=x86_64
build_type=Release
compiler=apple-clang
compiler.libcxx=libc++
compiler.version=13.1
os=Macos
os.version=10.14
[options]
[build_requires]
[env]

create-dmg/1.1.0: Forced build from source
create-dmg/1.1.0 (test package): Installing package
Requirements
    create-dmg/1.1.0 from local cache - Cache
Packages
    create-dmg/1.1.0:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 - Build

Installing (downloading, building) binaries...
create-dmg/1.1.0: WARN: Build folder is dirty, removing it: /Users/martin/.conan/data/create-dmg/1.1.0/_/_/build/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9
create-dmg/1.1.0: Configuring sources in /Users/martin/.conan/data/create-dmg/1.1.0/_/_/source
create-dmg/1.1.0: Copying sources to build folder
create-dmg/1.1.0: Building your package in /Users/martin/.conan/data/create-dmg/1.1.0/_/_/build/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9
create-dmg/1.1.0: Generator txt created conanbuildinfo.txt
create-dmg/1.1.0: Aggregating env generators
create-dmg/1.1.0: Calling build()
Downloading v1.1.0.tar.gz completed [44.58k]

create-dmg/1.1.0:
create-dmg/1.1.0: ERROR: Package '5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9' build failed
create-dmg/1.1.0: WARN: Build folder /Users/martin/.conan/data/create-dmg/1.1.0/_/_/build/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9
ERROR: create-dmg/1.1.0: Error in build() method, line 29
	tools.patch(**patch)
	FileNotFoundError: [Errno 2] No such file or directory: 'patches/0001-change-share-to-res.patch'
make: *** [test] Error 1
```

**Review by uilianries — APPROVED** [review]
LGTM

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**